crates/sncast/src/lib.rs

```diff
@@ -75,8 +76,54 @@ impl NumbersFormat {
 
 pub struct WaitForTx {
     pub wait: bool,
-    pub timeout: u16,
-    pub retry_interval: u8,
+    pub wait_params: ValidWaitParams,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, Copy, PartialEq)]
+pub struct ValidWaitParams {
```

Comment:
Let's just rename it to WaitParams, as it does not make sense to pass invalid params :)

Reply:
maybe `ValidatedWaitParams`? :)

Reply:
Still a bit redundant, but good enough for me :+1:

Reply:
I just think that ValidatedWaitParams gives a signal that it's not just a struct collecting `retry_interval` and `timeout`, but a struct that also assures that these params are valid